pkg/client/kubernetes/client.go

```diff
@@ -82,6 +84,9 @@ func NewClientFromSecretObject(secret *corev1.Secret) (Client, error) {
 }
 
 func newClientSet(config *rest.Config, clientConfig clientcmd.ClientConfig) (Client, error) {
+	config.Dial = (&net.Dialer{Timeout: 5 * time.Second}).DialContext
```

Comment:
Maybe is better to externalize those settings in https://github.com/gardener/gardener/blob/937b603f5ee8cded2140fb3a7e6253eb1a076f26/pkg/apis/componentconfig/v1alpha1/types.go#L54-L71 

They can be a field of type `metav1.Duration`